io.go

```diff
@@ -44,6 +47,42 @@ func (cr *CborReader) SetReader(r io.Reader) {
 	cr.r = GetPeeker(r)
 }
 
+func (cr *CborReader) ReadCid(scratchBuf []byte) (cid.Cid, error) {
+	maj, extra, err := cr.ReadHeader()
+	if err != nil {
+		if err == io.EOF {
+			err = io.ErrUnexpectedEOF
+		}
+		return cid.Undef, err
+	}
+
+	if maj != MajTag {
+		return cid.Undef, fmt.Errorf("expected cbor type 'tag' in input")
+	}
+
+	if extra != 42 {
+		return cid.Undef, fmt.Errorf("expected tag 42")
+	}
+
+	if extra > 512 {
+		return cid.Undef, fmt.Errorf("header size too big for a cid")
+	}
+
+	if len(scratchBuf) < int(extra) {
+		return cid.Undef, fmt.Errorf("scratchBuf not large enough for cid")
```

Comment:
Don't we usually allocate header-sized scratch bufs?

Reply:
yeah, this is actually a different path entirely (not used *in* codebase yet). I wanted this for doing some manual/optimized reading of certain objects